packages/framework/ai-collab/README.md

```diff
@@ -238,10 +239,166 @@ To see detailed information about each event, please read their cooresponding [t
+
+Lets look at an example of the Insert Diff
+The following `InsertDiff` is an example of a `Diff` that would result from if the ai agent inserts an object into index 1 of `TestAppRootObject.rootVectors`
+### Example Insert Ui Diff
+```json
+type: "insert",
+nodePath: [
+		{
+			shortId: -14,
+			schemaIdentifier: "testApp.Todo",
+			parentField: 1,
+		},
+		{
+			shortId: undefined,
+			schemaIdentifier: "testApp.Array<[\"testApp.Todo\"]>",
+			parentField: "todos",
+		},
+		{
+			shortId: -1,
+			schemaIdentifier: "testApp.TestAppRootObject",
+			parentField: "rootFieldKey",
+		},
+	],
+  	aiExplanation: "I need to insert a todo within the todos array",
+  	nodeContent: {
+		id: "f75951b0-df9d-4daa-aaf2-c322f2f462a8",
+		title: "Example Todo Title",
+		description: "Example Todo Description"
+	},
+```
+As you can see, the object at the beginning of the `nodePath` array directly points to the newly inserted node while each next object is the parent of the preceding node until you hit the root node passed to the ai-collab function call.
+
+Using the following ui diff, you can identify the modified node in a number of different ways.
+
+The simplest way is to use the `shortId` where you can use the following code to identify the newly inserted node within the SharedTree.
+
+**Note that the shortId field will only exist for objects that have a field defined as the `SchemaFactory.identifier` field.** See the above example app schema in this section to see the schema field defined as `sf.identifier`
+
+Lets take a look at another UI example of using an array of Diffs to render changes
+```ts
+import { Tree } from "@fluidframework/tree/alpha"
+
+function renderTodoWithDiffs(todo: Todo, Diffs: diff[]) {
+const modifyDiffs = Diffs.filter((diff): diff is ModifyDiff => diff.type === "modify") ?? [];
+const matchingModifyDiffs = modifyDiffs.filter(
+	(diff: ModifyDiff) =>
+		// Modify diffs are a field level edit, so the first path will be the field on the target node and the second will be the node itself.
+		diff.nodePath.length > 1 && diff.nodePath[1]?.shortId === Tree.shortId(task),
+);
+
+const insertDiffs = Diffs.filter((diff): diff is InsertDiff => diff.type === "insert") ?? [];
+const matchingInsertDiffs = insertDiffs.filter(
+	(diff: InsertDiff) =>
+		// Insert diffs are a node level edit, so the first path will be the node.
+		diff.nodePath[0]?.shortId === Tree.shortId(task),
+);
+}
+
+if (insertDiffs.length > 0) {
+	renderNewlyInsertedTodo(todo)
+} else if (modifyDiffs.length > 0) {
+	const modifiedFields: [] = matchingModifyDiffs.map(diff => diff.nodePath[0].parentField);
+	renderModifiedTodo(todo, modifiedFields)
+} else {
+	renderTodo(todo)
+}
+
+```
+You can also use the `type` and `schemaIdentifier` fields to group related `diff`'s
+```ts
```

Comment:
Nit: there should be blank lines before and after fenced code blocks. GitHub won't let me make the suggestion because all of the backticks confuse it 🤣